src/index.rs

```diff
@@ -291,41 +296,95 @@ impl IndexRead {
     }
 
     pub(crate) fn open(transport: Arc<dyn Transport>) -> IndexRead {
-        IndexRead { transport }
+        IndexRead {
+            transport,
+            decompressor: Decompressor::new(),
+            stats: IndexReadStats::default(),
+        }
     }
 
-    /// Make an iterator that will return all entries in this band.
-    pub fn iter_entries(self) -> IndexEntryIter<IndexHunkIter> {
-        // TODO: An option to pass in a subtree?
-        IndexEntryIter::new(self.iter_hunks(), Apath::root(), Exclude::nothing())
+    /// Clone the read index.
+    /// Note:
+    /// This has several side effects:
+    /// - Depending on the implementation of the decompressor, duplicate might not be a cheap option.
+    /// - Every read index has its own unique read stats, therefore the clone does not inherit the read stats.
+    pub(crate) fn duplicate(&self) -> Self {
```

Comment:
I don't think the decompressor would ever be expensive: it's basically a buffer?

Maybe this could just be `impl Clone`?

Reply:
I agree, that the argument of not using `Clone` based complexity of cloning the decoder in itself wouldn't be enough. But we got two side effects here, which I've mentioned in the doc-comment in the code above seem severe enough that I would argue it's not actually creating a clone.